### PR TITLE
fix: remove duplicated keyword argument `timeout`

### PR DESCRIPTION
**Pull request by m9810223**
Traceback:
```
    await page.locator('#...').screenshot(
  File "/usr/lib/python3.8/site-packages/playwright/async_api/_generated.py", line 16317, in screenshot
    await self._impl_obj.screenshot(
  File "/usr/lib/python3.8/site-packages/playwright/_impl/_locator.py", line 479, in screenshot
    return await self._with_element(
  File "/usr/lib/python3.8/site-packages/playwright/_impl/_locator.py", line 105, in _with_element
    return await task(
  File "/usr/lib/python3.8/site-packages/playwright/_impl/_locator.py", line 480, in <lambda>
    lambda h, timeout: h.screenshot(timeout=timeout, **params)
TypeError: screenshot() got multiple values for keyword argument 'timeout'
```

I fix this problem with the coding style:
https://github.com/microsoft/playwright-python/blob/47caa7c1d01b2eef6617a5d24513f8c07449a765/playwright/_impl/_locator.py#L395-L400

### CODE REVIEW

File: playwright/_impl/_locator.py [hunk 1]
```diff
@@ -517,6 +517,7 @@ async def screenshot(
         mask: List["Locator"] = None,
     ) -> bytes:
         params = locals_to_params(locals())
+        del params["timeout"]
         return await self._with_element(
             lambda h, timeout: h.screenshot(timeout=timeout, **params)
```
Comment (mxschmitt) [inline playwright/_impl/_locator.py:522]:
```suggestion
            lambda h, timeout: h.screenshot(**params)
```

do you mind adding a test for it? I think also this is a cleaner bugfix.

Comment (m9810223) [reply playwright/_impl/_locator.py:522]: Sure. I can add the test.

But I have one question about your solution:

- The `lambda function` also has an arg: `timeout`. If you remove `timeout=timeout` that means `timeout` arg in the lambda function is uesless. 
  So which (variable) value these methods should revieved?

```py
    async def select_text(self, force: bool = None, timeout: float = None) -> None:
        #                                           [m-arg]
        params = locals_to_params(locals())
        del params["timeout"] # [PR]
        return await self._with_element(
            lambda h, timeout: h.select_text(timeout=timeout, **params), timeout
        #             [f-arg]                        [f-arg?]    [PR]    [m-arg?]
        )

    # m-arg -> method arg
    # f-arg -> lambda function arg
```

**Comment by m9810223** [comment]
Hi @mxschmitt,

Sure. I can add the test.

But I have one question about your solution:

- The `lambda function` also has an arg: `timeout`. If you remove `timeout=timeout` that means `timeout` arg in the lambda function is uesless. 
  So which (variable) value these methods should revieved?

```py
    async def select_text(self, force: bool = None, timeout: float = None) -> None:
        #                                           [m-arg]
        params = locals_to_params(locals())
        del params["timeout"] # [PR]
        return await self._with_element(
            lambda h, timeout: h.select_text(timeout=timeout, **params), timeout
        #             [f-arg]                        [f-arg?]    [PR]    [m-arg?]
        )

    # m-arg -> method arg
    # f-arg -> lambda function arg
```

**Comment by m9810223** [comment]
Hi @mxschmitt,

Sure. I can add the test.

But I have one question about your solution:

- The `lambda function` also has an arg: `timeout`. If you remove `timeout=timeout` that means `timeout` arg in the lambda function is uesless. 
  So which (variable) value these methods should revieve?

```py
    async def select_text(self, force: bool = None, timeout: float = None) -> None:
        #                                           [m-arg]
        params = locals_to_params(locals())
        del params["timeout"] # [PR]
        return await self._with_element(
            lambda h, timeout: h.select_text(timeout=timeout, **params), timeout
        #             [f-arg]                        [f-arg?]    [PR]    [m-arg?]
        )

    # m-arg -> method arg
    # f-arg -> lambda function arg
```

**Comment by mxschmitt** [comment]
> The lambda function also has an arg: timeout. If you remove timeout=timeout that means timeout arg in the lambda function is useless.

You are right, I didn't realise that. It should receive of course the lambda timeout.

I think my favourite fix would be: `h.select_text(**{**params, "timeout": timeout}),` which makes it clear that we have a timeout override instead of a random property deletion.

**Comment by m9810223** [comment]
Cool! I think dict unpacking style is better than using the `del` syntax, too.

**Comment by m9810223** [comment]
I use `**dict(**params)` instead of `**{**params}` to pass mypy test.
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[Literal['jpeg', 'png']]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Union[str, Path, None]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[int]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[bool]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[Literal['allow', 'disabled']]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[Literal['hide', 'initial']]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[Literal['css', 'device']]"
playwright/_impl/_locator.py:521: error: Argument 1 to "screenshot" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[List[Locator]]"
playwright/_impl/_locator.py:553: error: Argument 1 to "select_text" of "ElementHandle" has incompatible type "**Dict[str, float]"; expected "Optional[bool]"
Found 9 errors in 1 file (checked 1 source file)
```

**Review by mxschmitt — APPROVED** [review]

**Comment by mxschmitt** [comment]
Thanks!

**Pull request merged**